src/realm/exceptions.hpp

```diff
@@ -176,6 +176,25 @@ class NoSubscriptionForWrite : public std::runtime_error {
     NoSubscriptionForWrite(const std::string& msg);
 };
 
+namespace query_parser {
```

Comment:
I should have put my previous comment on naming here. Since these exceptions are thrown from the query engine now and not necessarily from the query parser, can we change this namespace to something more general such as `query`?

Reply:
Let's not change this in this PR.

Reply:
Ok sure. In fact, I'd expect all these exceptions will changing soon with the upcoming error handling work.